app/controllers/buses_controller.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+require 'json'
+
+class BusesController < ApplicationController
+  def index
+  end
+
+  def stop
+    stop = params[:stop]
+    @arrivals = fetch_arrivals([stop])
+    render :arrivals
+  end
+
+  def postcode
+    postcode = params[:postcode]
+    lat, lon = fetch_postcode_coordinates(postcode)
+    stops = fetch_nearest_stops(lat, lon)
+    @arrivals = fetch_arrivals(stops)
+    render :arrivals
+  end
+
+  private
+
+  def fetch_arrivals(stops)
+    arrivals = []
+
+    stops.each do |stop|
+      uri = URI("https://api.tfl.gov.uk/StopPoint/#{stop}/Arrivals")
+      res = Net::HTTP.get(uri)
+      buses = JSON.parse(res)
+      buses
+        .sort_by! { |x| x['timeToStation'].to_i }
+        .slice(0, 5)
+      arrivals.push(buses)
+    end
+
+    arrivals
```

Comment:
I'd prefer `return arrivals` for clarity but I don't know what the linter has to say about that

Reply:
The linter suggests no return since ruby implicitly returns the final statement of a function anyway